src/main.js

```diff
@@ -285,23 +297,29 @@ function handleClose(streams, children, childrenInfo) {
         if (aliveChildren.length === 0) {
             exit(exitCodes);
         }
-
-        if (config.killOthers && !othersKilled) {
-            othersKilled = true
+        if (!othersKilled) {
+          othersKilled = true;
```

Comment:
By updating the flag here, you're preventing `--kill-others-on-fail` to work when the first process exists with success